File: 1801.cc
#include "iostream"
#include "map"
#include "vector"

int	main()
{
	int res = 0, n;
	int res2 = 0;
	std::map<int, int> M;
	std::vector<int> A;
	while (std::cin >> n)
	{
		res += n;
		A.push_back(n);
	}
	std::cout << res << std::endl;
	int i = 0, count = 0;
	res = 0;

	M[0] = 1;
	while (1)
	{
		// std::cout << i << ' ' << A[i] << ' ' << res << std::endl;
		res += A[i];
		++M[res];
		if (M[res] == 2)
			break ;
		i = i == A.size() - 1 ? 0 : i + 1;
		++count;
	}
	std::cout << res << " at " << count << std::endl;
}

File: 1801.py
file = open('_inputs_/1801.0')
a = []
for line in file:
    a.append(line)
res = 0
for line in a:
    res += eval(line)
res2 = 0
S = set()
i = 0
while i < len(a):
    if res2 in S:
        break
    S.add(res2)
    res2 += eval(a[i])
    i += 1
    if i == len(a):
        i = 0
print(res)
print(res2)
assert res == 529
assert res2 == 464
